bootstrap/kubeadm/api/v1alpha3/kubeadmbootstrapconfig_types.go

```diff
@@ -175,6 +175,98 @@ type File struct {
 	Content string `json:"content"`
 }
 
+// FileSource is a union of all possible external source types for file data.
+// Only one field may be populated in any given instance. Developers adding new
+// sources of data for target systems should add them here.
+type FileSource struct {
+	// Raw represents raw user-provided string content to populate this file.
+	Raw *File `json:"raw,omitempty"`
+
+	// Secret represents a secret that should populate this file.
+	// +optional
+	Secret *SecretFileSource `json:"secret,omitempty"`
```

Comment:
I'm a little bit confused by this, because SecretFileSource is a specialization of  FileSource, but it seems that it allows acting as a source for many files (like a mount)

Reply:
Correct, based on my reading of https://github.com/kubernetes-sigs/cluster-api/pull/1860#discussion_r355594849 and similar, this is desired behavior.

The thinking was that a user would specify the source one time, but could project multiple files (keys in the data of the secret). If you specify raw content, the file source can only produce one output.